src/server/rdb_load.cc

```diff
@@ -2529,6 +2530,16 @@ void RdbLoaderBase::CopyStreamId(const StreamID& src, struct streamID* dest) {
   dest->seq = src.seq;
 }
 
+bool RdbLoaderBase::ShouldReuse(const RdbVariant& var) {
+  return visit(
+      Overloaded{[](long long) { return true; },
+                 [](const LzfString& lzf) { return lzf.compressed_blob.capacity() < 1_MB; },
+                 [](const unique_ptr<LoadTrace>& ptr) { return false; },
+                 [](const base::PODArray<char>& arr) { return arr.capacity() < 1_MB; },
```

Comment:
I see we are not reusing the item value 
inside ReadObj we do 
*dest = std::move(*iores);
So what is actually the value of using item_queue and only for specific datatypes with specific capacity? we can add monostate to this RdbVariant and reset it before we push to item_queue

Reply:
you are right. I do not remember why I did it but I suspect that the goal of item_queue was to reduce the pressure from memory allocator that got really confused when we we  allocated LOTS of data in the parsing thread and deallocated it in the object creation thread.